### PR TITLE
KUBESAW-12: Convert the health-check goroutine into ToolchainCluster controller

### PR DESCRIPTION
**Pull request by fbm3307**
This is to convert the health-check goroutine into ToolchainCluster controller

The related PRs:

- Toolchain-Common - https://github.com/codeready-toolchain/toolchain-common/pull/386
- Host-Operator - https://github.com/codeready-toolchain/host-operator/pull/1010

The Other PRs where only Toolchain common dependency is updated 
- Toolchain-e2e - https://github.com/codeready-toolchain/toolchain-e2e/pull/949
- Registration-service - https://github.com/codeready-toolchain/registration-service/pull/422
- Sandbox-sre - https://github.com/codeready-toolchain/sandbox-sre/pull/1698

### CODE REVIEW

**Review by filariow — APPROVED** [review]

**Review by mfrancisc — APPROVED** [review]

File: main.go [hunk 1]
```diff
@@ -222,6 +226,14 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "ToolchainClusterCache")
 		os.Exit(1)
 	}
+
+	if err := toolchainclusterhealth.NewReconciler(
+		mgr,
+		requeAfter,
+	).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ToolchainClusterHealth")
```
Comment (MatousJobanek) [inline main.go:234]:
```suggestion
		setupLog.Error(err, "unable to create controller", "controller", "ToolchainCluster")
```

File: main.go [hunk 2]
```diff
@@ -222,6 +226,14 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "ToolchainClusterCache")
 		os.Exit(1)
 	}
+
+	if err := toolchainclusterhealth.NewReconciler(
+		mgr,
+		requeAfter,
```
Comment (MatousJobanek) [inline main.go:232]: the same as in host-operator PR

**Review by MatousJobanek — COMMENTED** [review]
Please, update the dependency to toolchain-common PR here as well

File: main.go (hunk 2 above)
Comment (fbm3307) [reply main.go]: done

File: main.go (hunk 1 above)
Comment (fbm3307) [reply main.go]: done

**Review by rajivnathan — APPROVED** [review]

**Pull request merged**